lib/ansible/modules/cloud/azure/azure_rm_appserviceplan.py

```diff
@@ -0,0 +1,376 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Yunge Zhu, <yungez@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_appserviceplan
+version_added: "2.7"
+short_description: Manage App Service Plan
+description:
+    - Create, update and delete instance of App Service Plan.
+
+options:
+    resource_group:
+        description:
+            - Name of the resource group to which the resource belongs.
+        required: True
+
+    name:
+        description:
+            - Unique name of the app service plan to create or update.
+        required: True
+
+    location:
+        description:
+            - Resource location. If not set, location from the resource group will be used as default.
+
+    sku:
+        description:
+            - The pricing tiers, e.g., F1, D1, B1, B2, B3, S1, P1, P1V2 etc.
+            - Please see https://azure.microsoft.com/en-us/pricing/details/app-service/plans/ for more detail.
+            - For linux app service plan, please see https://azure.microsoft.com/en-us/pricing/details/app-service/linux/ for more detail.
+
+    is_linux:
+        description:
+            - Descirbe whether to host webapp on Linux worker.
+        type: bool
+        default: false
+
+    number_of_workers:
+        description:
+            - Describe number of workers to be allocated.
+
+    state:
+      description:
+        - Assert the state of the app service plan.
+        - Use 'present' to create or update an app service plan and 'absent' to delete it.
+      default: present
+      choices:
+        - absent
+        - present
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Yunge Zhu(@yungezz)"
+
+'''
+
+EXAMPLES = '''
+    - name: Create a windows app service plan
+      azure_rm_appserviceplan:
+        name: "{{ win_plan_name }}1"
+        resource_group: "{{ plan_resource_group }}"
+        location: "{{ location }}"
+        sku: S1
+
+    - name: Create a linux app service plan
+      azure_rm_appserviceplan:
+        resource_group: "{{ linux_plan_resource_group }}"
+        name: "{{ linux_plan_name }}1"
+        location: "{{ location }}"
+        sku: S1
+        is_linux: true
+        number_of_workers: 1
+
+    - name: update sku of existing windows app service plan
+      azure_rm_appserviceplan:
+        name: "{{ win_plan_name }}1"
+        resource_group: "{{ plan_resource_group }}"
+        location: "{{ location }}"
+        sku: S2
+'''
+
+RETURN = '''
+azure_appserviceplan:
```

Comment:
Should this module be returning all this information? Usually we do that for facts modules instead. If you want to keep some values they should just be on the root of the json not in `azure_appserviceplan` as well as documented properly and not shown in the sample.

Reply:
fixed to return id.